crates/matrix-sdk-indexeddb/src/state_store/migrations.rs

```diff
@@ -499,6 +507,99 @@ async fn migrate_to_v5(
     })
 }
 
+/// Remove the old user IDs stores and populate the new ones.
+async fn migrate_to_v6(
+    db: &IdbDatabase,
+    store_cipher: Option<&StoreCipher>,
+) -> Result<OngoingMigration> {
+    // We only have joined and invited user IDs in the old store, so instead we will
+    // use the room member events to populate the new store.
+    let tx = db.transaction_on_multi_with_mode(
+        &[keys::ROOM_STATE, keys::ROOM_INFOS, keys::STRIPPED_ROOM_STATE, keys::STRIPPED_ROOM_INFOS],
+        IdbTransactionMode::Readonly,
+    )?;
+
+    let state_store = tx.object_store(keys::ROOM_STATE)?;
+    let room_infos = tx
+        .object_store(keys::ROOM_INFOS)?
+        .get_all()?
+        .await?
+        .iter()
+        .filter_map(|f| deserialize_event::<RoomInfo>(store_cipher, f).ok())
+        .collect::<Vec<_>>();
+    let mut values = Vec::new();
+
+    for room_info in room_infos {
+        let room_id = room_info.room_id();
+        let range =
+            encode_to_range(store_cipher, keys::ROOM_STATE, (room_id, StateEventType::RoomMember))?;
+        for value in state_store.get_all_with_key(&range)?.await?.iter() {
+            let member_event =
+                deserialize_event::<Raw<SyncRoomMemberEvent>>(store_cipher, value.clone())?
```

Comment:
Is it really necessary to clone `value` before deserialization here?

Reply:
Well according to the `deserialize_event` function signature, yes. According to what the function does, it could only take a reference.

Reply:
Okay, if you want change it here, but we can do it in a follow-up PR as well.

Reply:
I'd rather do a follow-up PR